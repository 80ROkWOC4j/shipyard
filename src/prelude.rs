pub use crate::get::GetComponent;
pub use crate::iter::{iterators, IntoIter, Shiperator};
pub use crate::not::Not;
pub use crate::remove::Remove;
pub use crate::run::System;
#[doc(hidden)]
pub use crate::run::SystemData;
pub use crate::sparse_set::{sort, sort::Sortable, View, ViewMut};
pub use crate::storage::{AllStorages, Entities, EntitiesMut, Key};
pub use crate::world::World;
pub use crate::Unique;
#[doc(hidden)]
#[cfg(feature = "proc")]
pub use shipyard_proc::system;

#[cfg(feature = "parallel")]
pub use crate::ThreadPool;
