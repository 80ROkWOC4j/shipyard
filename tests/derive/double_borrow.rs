use shipyard::prelude::*;

#[system(Double)]
fn run(_: &u32, _: &mut u32) {}

#[system(DoubleUnique)]
fn run(_: &u32, _: Unique<&mut u32>) {}

#[system(UniqueDouble)]
fn run(_: Unique<&u32>, _: &mut u32) {}

#[system(UniqueUnique)]
fn run(_: Unique<&u32>, _: Unique<&mut u32>) {}

fn main() {}
