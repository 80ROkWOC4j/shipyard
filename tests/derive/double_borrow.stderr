error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
 --> $DIR/double_borrow.rs:4:8
  |
4 | fn run(_: &u32, _: &mut u32) {}
  |        ^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
 --> $DIR/double_borrow.rs:7:8
  |
7 | fn run(_: &u32, _: Unique<&mut u32>) {}
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
  --> $DIR/double_borrow.rs:10:8
   |
10 | fn run(_: Unique<&u32>, _: &mut u32) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
  --> $DIR/double_borrow.rs:13:8
   |
13 | fn run(_: Unique<&u32>, _: Unique<&mut u32>) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
